pennylane/templates/subroutines/prepselprep.py

```diff
@@ -0,0 +1,164 @@
+    """Compute a new sum of unitaries with positive coefficients"""
+
+    new_coeffs = []
+    new_ops = []
+
+    for coeff, op in zip(*lcu.terms()):
+
+        angle = qml.math.angle(coeff)
+        new_coeffs.append(qml.math.abs(coeff))
+
+        new_op = op @ qml.GlobalPhase(-angle, wires=op.wires)
+        new_ops.append(new_op)
+
+    interface = qml.math.get_interface(lcu.terms()[0])
+    new_coeffs = qml.math.array(new_coeffs, like=interface)
+
+    return new_coeffs, new_ops
+
+
+class PrepSelPrep(Operation):
+    """This class implements a block-encoding of a linear combination of unitaries
+    using the Prepare, Select, Prepare method"""
+
+    def __init__(self, lcu, control=None, id=None):
+        coeffs, ops = lcu.terms()
+        control = qml.wires.Wires(control)
+        self.hyperparameters["lcu"] = lcu
+        self.hyperparameters["coeffs"] = coeffs
+        self.hyperparameters["ops"] = ops
+        self.hyperparameters["control"] = control
+
+        if any(
+            control_wire in qml.wires.Wires.all_wires([op.wires for op in ops])
+            for control_wire in control
+        ):
+            raise ValueError("Control wires should be different from operation wires.")
+
+        target_wires = qml.wires.Wires.all_wires([op.wires for op in ops])
+        self.hyperparameters["target_wires"] = target_wires
+
+        all_wires = target_wires + control
+        super().__init__(*self.data, wires=all_wires, id=id)
+
+    def _flatten(self):
+        return tuple(self.lcu), (self.control)
+
+    @classmethod
+    def _unflatten(cls, data, metadata) -> "PrepSelPrep":
+        coeffs = [term.terms()[0][0] for term in data]
+        ops = [term.terms()[1][0] for term in data]
+        lcu = qml.ops.LinearCombination(coeffs, ops)
+        return cls(lcu, metadata)
+
+    def __repr__(self):
+        return f"PrepSelPrep(coeffs={tuple(self.coeffs)}, ops={tuple(self.ops)}, control={self.control})"
+
+    def map_wires(self, wire_map: dict) -> "PrepSelPrep":
+        new_ops = [o.map_wires(wire_map) for o in self.hyperparameters["ops"]]
+        new_control = [wire_map.get(wire, wire) for wire in self.hyperparameters["control"]]
+        new_lcu = qml.ops.LinearCombination(self.hyperparameters["coeffs"], new_ops)
+        return PrepSelPrep(new_lcu, new_control)
+
+    def decomposition(self):
+        return self.compute_decomposition(self.lcu, self.control)
+
+    @staticmethod
+    def compute_decomposition(lcu, control):
+        coeffs, ops = _get_new_terms(lcu)
+
+        decomp_ops = []
+        decomp_ops.append(
+            qml.AmplitudeEmbedding(qml.math.sqrt(coeffs), normalize=True, pad_with=0, wires=control)
+        )
+        decomp_ops.append(qml.Select(ops, control))
+        decomp_ops.append(
+            qml.adjoint(
+                qml.AmplitudeEmbedding(
+                    qml.math.sqrt(coeffs), normalize=True, pad_with=0, wires=control
+                )
+            )
+        )
```

Comment:
You can probably just add them to the list directly.

```suggestion
        decomp_ops = [
            qml.AmplitudeEmbedding(qml.math.sqrt(coeffs), normalize=True, pad_with=0, wires=control)
        ), 
        qml.Select(ops, control), 
        qml.adjoint(qml.AmplitudeEmbedding(
            qml.math.sqrt(coeffs), normalize=True, pad_with=0, wires=control
                )
            ),
        ]
```

Reply:
Thanks, I just pushed this change.